### DISCUSSION TITLE
Upload files with specific file extensions

### DISCUSSION THREAD

**Discussion by graysuit** [community/community#27087] [Actions]
<p>As previously I opened an issue at: <a href="https://github.com/actions/upload-artifact/issues/178" class="inline-onebox" rel="noopener nofollow ugc">upload files with specific file extensions · Issue #178 · actions/upload-artifact · GitHub</a><br>
But answer doesn’t seems to be working.</p>
<p>Answer was: <code>path: **/*.exe</code><br>
But says <code>undefined alias path: **/*.exe</code></p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/8/3/83a32c3b1797975c27c55e03514f7e0159b46629.png" data-download-href="/uploads/short-url/iMw8YbNUIgbnnWTedxxxN9isN29.png?dl=1" title="1" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/54432616/181101623-4854d5a8-1f85-4f02-b423-6e5958532f35.png" alt="1" data-base62-sha1="iMw8YbNUIgbnnWTedxxxN9isN29" width="690" height="246" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/8/3/83a32c3b1797975c27c55e03514f7e0159b46629_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">1</span><span class="informations">900×321 15.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div><br>
<div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/3/6/366a72a06c2db95870bc374370c99ced64f96e27.png" data-download-href="/uploads/short-url/7LnQuDjfZtpJa3gA86a6RZ6p23J.png?dl=1" title="2" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/54432616/181101624-0abf32d2-0e40-42f2-9f44-f949aeafaa65.png" alt="2" data-base62-sha1="7LnQuDjfZtpJa3gA86a6RZ6p23J" width="690" height="360" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/3/6/366a72a06c2db95870bc374370c99ced64f96e27_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">2</span><span class="informations">839×438 10.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by airtower-luna** [comment]
<p><code>*</code> in YAML references an anchor, e.g. <code>*A</code> references whatever was defined with <code>&amp;A</code>. Add quotes to explicitly make the value a string:</p>
<pre><code class="lang-yaml">path: '**/*.exe'
</code></pre>

**Comment by graysuit** [comment]
<p>Hi sir,<br>
thanks for jet reply.</p>
<p>I have set as advised, But no luck.</p>
<pre><code class="lang-auto">    - name: Upload Artifact
      uses: actions/upload-artifact@v1.0.0
      with:
       name: Releases
       path: '**/*.exe'
</code></pre>
<p>Please see my full yml file: <a href="https://github.com/DarkSecDevelopers/RunSomeAware/blob/5ec6f148a9243a21c828202c78a7a66f33ac1427/.github/workflows/Csharp_Compiler.yml#L22" class="inline-onebox" rel="noopener nofollow ugc">RunSomeAware/Csharp_Compiler.yml at 5ec6f148a9243a21c828202c78a7a66f33ac1427 · DarkSecDevelopers/RunSomeAware · GitHub</a></p>
<p>Error msg:</p>
<pre><code class="lang-auto"> Error: Path does not exist D:\a\RunSomeAware\RunSomeAware\**\*.exe
Error: Exit code 1 returned from process: file name 'c:\runners\2.278.0\bin\Runner.PluginHost.exe', arguments 'action "GitHub.Runner.Plugins.Artifact.PublishArtifact, Runner.Plugins"'.
</code></pre>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">

      <a href="https://github.com/DarkSecDevelopers/RunSomeAware/runs/2810743495?check_suite_focus=true#step:4:9" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/54432616/181101628-7e6d2690-0ca0-406a-a2a5-6dae19d39e38.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://github.com/DarkSecDevelopers/RunSomeAware/runs/2810743495?check_suite_focus=true#step:4:9" target="_blank" rel="noopener nofollow ugc">Build software better, together</a></h3>

  <p>GitHub is where people build software. More than 65 million people use GitHub to discover, fork, and contribute to over 200 million projects.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by jsoref** [comment]
<p>My guess is that this feature only works on Unix shells. (i.e. it’s depending on the shell to do the expansion.)</p>
<p>I’d suggest filing <em>another</em> bug. The inconsistent behavior is really unacceptable.</p>
<p>(Sorry, I don’t have the energy to file this bug. But, you should try to be more polite than I would be when you file it.)</p>

**Comment by graysuit** [comment]
<p>Are you sure its a bug ?<br>
And what you mean by <code>inconsistent behavior</code> ?</p>

**Comment by jsoref** [comment]
<p>80% certain (this is fairly low for me).</p>
<p>Inconsistent in that it does one thing on macOS and Linux, and a different thing on Windows.</p>
<p>It’s generally reasonable to expect the same behavior on all supported platforms unless otherwise documented, and I haven’t seen any documentation saying otherwise.</p>

**Comment by Simran-B** [comment]
<p>It’s a Windows runner and the workflow uses the default shell, which is <code>pwsh</code> in this case. PowerShell Core appears to support path expansion. <code>ls **\*.exe</code> worked in my test. However, judging by the error message, the shell does not appear to be involved at all. Whatever tries to work with the supplied path <code>**/*.exe</code> manages to turn the forward into a backward slash, but then seems to interpret the path literally instead of expanding it to just files with the extension <code>.exe</code> from any subfolders.</p>
<p>The expectation is that the glob pattern <code>**/*.exe</code> should work across all runner operating systems. The readme of <a href="https://github.com/actions/upload-artifact" rel="noopener nofollow ugc">actions/upload-artifact</a> states that it uses <code>@actions/glob</code> for wildcard matching. In turn, its readme indirectly states that it supports Windows and there’s no remark about issues with globbing on Windows: <a href="https://github.com/actions/toolkit/tree/main/packages/glob#glob-behavior" class="inline-onebox" rel="noopener nofollow ugc">toolkit/packages/glob at main · actions/toolkit · GitHub</a><br>
I even tried it out in Node.js and it appears to work fine:</p>
<pre><code class="lang-auto">const glob = require("@actions/glob")
glob.create("C:/path/to/**/*.exe").then(g =&gt; g.glob().then(r =&gt; console.dir(r)))
/*
  [
    "C:\\path\\to\\bin\\app.exe",
    ...
  ]
*/
</code></pre>

**Comment by Simran-B** [comment, marked as answer]
<p>Looking at the workflow, I realized that it uses a super outdated version of the <code>actions/upload-artifacts</code> action. It probably doesn’t support globbing.</p>
<pre><code class="lang-yaml">    - name: Upload Artifact
      uses: actions/upload-artifact@v1.0.0
      with:
       name: Releases
       path: '**/*.exe'
</code></pre>
<p>Replace it with the following code that changes the version to <code>v2</code> (and fixes the indentation):</p>
<pre><code class="lang-yaml">    - name: Upload Artifact
      uses: actions/upload-artifact@v2
      with:
        name: Releases
        path: '**/*.exe'
</code></pre>

**Comment by graysuit** [comment]
<p>Thanks so much. You are professional. It worked.<br>
I was using old artifact version.<br>
Now it works fine: <a href="https://github.com/DarkSecDevelopers/RunSomeAware/runs/2819734408?check_suite_focus=true#step:4:9" class="inline-onebox" rel="noopener nofollow ugc">Update Csharp_Compiler.yml · DarkSecDevelopers/RunSomeAware@33be108 · GitHub</a></p>
<p>Cheers ! :heart:</p>
